### ISSUE TITLE
Unable to use file directives to include the contents of individual processors v0.109.0

### ISSUE THREAD

**Issue by arielvalentin** [open-telemetry/opentelemetry-collector#11154]
**Describe the bug**

In versions prior to 0.109.0 we were able to include the contents of a processor definition using a file directive, e.g.:

```yaml
transform/redaction-rules: ${file:redaction-rules.yaml}
```

However, when attempting to validate the configuration v0.109.0, it fails with the following error:

```yaml
error decoding 'processors': error reading configuration for "transform/redaction-rules:": unexpected sub-config value kind for key:transform/redaction-rules value:{map[error_mode:ignore....] error_mode: ignore
} kind:struct
```

**Steps to reproduce**

**What did you expect to see?**

I expect the behavior to remain the same, for the validation to succeed, and the collector to start up. 

**What did you see instead?**

The collector and/or validation commands fail with an error `error decoding 'processors': error reading configuration for "transform/redaction-rules:"`

**What version did you use?**

v0.109.0

**What config did you use?**

```yaml
service:
  pipelines:
    traces:
      receivers:
        - otlp
      processors:
        - batch
        - transform/redaction-rules
      exporters:
        - logging
        
# The pipeline details
receivers:
  otlp:
      http:

processors:
  transform/redaction-rules:
    ${file:redaction-rules.yaml}

  batch: {}

exporters:
  logging:
    verbosity: detailed
    sampling_initial: 1
    sampling_thereafter: 1

```

```yaml
 # transform/redaction-rules
 ---
error_mode: ignore
trace_statements:
- context: span
  statements:
    - |-
      set(attributes["http.target"],
      SHA256(attributes["http.target"]))
      where attributes["http.target"] != nil
    - |-
      set(attributes["http.url"],
      SHA256(attributes["http.url"]))
      where attributes["http.url"] != nil
    - |-
      set(attributes["url.full"],
      SHA256(attributes["url.full"]))
      where attributes["url.full"] != nil
```

**Environment**

**Additional context**

**Comment by mx-psi** [comment]
Thanks for reporting! I wrote a more minimal repro:
<details>
<summary>Minimal repro (click to expand)</summary>

```yaml
# config.yaml
receivers:
  nop: ${file:nop.yaml}

exporters:
  nop:

service:
  pipelines:
    traces:
      receivers: [nop]
      exporters: [nop]
```


```yaml
# nop.yaml
{}
```

Output:

```
❯ ./bin/otelcorecol_linux_amd64 --config config.yaml 
Error: failed to get config: cannot unmarshal the configuration: decoding failed due to the following error(s):

error decoding 'receivers': error reading configuration for "nop": unexpected sub-config value kind for key:nop value:{map[] # nop-config.yaml
{}
} kind:struct
2024/09/12 10:52:53 collector server run finished with error: failed to get config: cannot unmarshal the configuration: decoding failed due to the following error(s):

error decoding 'receivers': error reading configuration for "nop": unexpected sub-config value kind for key:nop value:{map[] # nop-config.yaml
{}
} kind:struct
```

</details>

This happens because of a bug on `confmap.Conf.Sub` that is leaking the internal representation of `confmap.Conf`. This happens since #10897 so I expect this is reproducible on v0.108.0 also. I will file a PR to fix this for v0.110.0.

In the meantime, an ugly but effective workaround should be to go one level higher/lower since the bug is specific for transclusion of a whole component config. For example, I believe the following should work:

```yaml
processors:
  transform/redaction-rules:
    error_mode: ignore
    trace_statements: ${file:file-with-only-the-rules.yaml}
```

**Comment by kago-dk** [comment]
I also ran into this issue.
